src/test/java/org/projectnessie/tools/catalog/migration/ITHiveAndNessie.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.catalog.migration;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.projectnessie.tools.catalog.migration.CatalogMigrationCLI.DRY_RUN_FILE;
+import static org.projectnessie.tools.catalog.migration.CatalogMigrationCLI.FAILED_IDENTIFIERS_FILE;
+import static org.projectnessie.tools.catalog.migration.CatalogMigrationCLI.FAILED_TO_DELETE_AT_SOURCE_FILE;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hive.HiveMetastoreTest;
+import org.apache.iceberg.nessie.NessieCatalog;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.testcontainers.containers.GenericContainer;
+
+public class ITHiveAndNessie extends HiveMetastoreTest {
+
+  private static String warehousePath1;
+
+  private static @TempDir File warehouse2;
+  private static String warehousePath2;
+
+  private static Catalog catalog1;
+
+  private static Catalog catalog2;
+
+  private static final Schema schema =
+      new Schema(Types.StructType.of(required(1, "id", Types.LongType.get())).fields());
+
+  private static final String IMAGE = "projectnessie/nessie:0.47.1";
+  private static final int NESSIE_PORT = 19121;
+
+  private static String nessieUri;
+
+  private static GenericContainer<?> container;
+
+  @BeforeAll
+  protected static void setup() throws Exception {
+    startMetastore();
+    warehousePath1 = catalog.getConf().get("hive.metastore.warehouse.dir");
+    warehousePath2 = String.format("file://%s", warehouse2.getAbsolutePath());
+
+    container =
+        new GenericContainer(IMAGE)
```

Comment:
We do have a Gradle plugin that starts Nessie as an external process, which should really be used instead of this manually crafted test-container handling, which is BTW also prone to leave running containers behind.